src/components/pagination.ts

```diff
@@ -0,0 +1,81 @@
+import {Component, EventEmitter, Input, Output, NgFor} from 'angular2/angular2';
+
+@Component({
+  selector: 'ngb-pagination',
+  directives: [NgFor],
+  template: `
+    <nav>
+      <ul class="pagination">
+        <li [class.disabled]="!hasPrevious()">
+            <a aria-hidden="true" (click)="selectPage(_page-1)">
+              <span aria-hidden="true">&laquo;</span>
+              <span class="sr-only">Previous</span>
+            </a>
+        </li>
+
+        <li *ng-for="#pageNumber of _pages" [class.active]="pageNumber === _page">
+          <a (click)="selectPage(pageNumber)">{{pageNumber}}</a>
+        </li>
+
+        <li [class.disabled]="!hasNext()">
+          <a aria-label="Next" (click)="selectPage(_page+1)">
+            <span aria-hidden="true">&raquo;</span>
+            <span class="sr-only">Next</span>
+          </a>
+        </li>
+      </ul>
+    </nav>
+  `
+})
+export class NgbPagination {
+  _collectionSize = 0;
+  _page = 0;
+  _pageSize = 10;
+  _pages: number[] = [];
+
+  @Input()
+  set page(newPage: number | string) {
+    this.selectPage(parseInt(`${newPage}`, 10));
```

Comment:
Oh, I was _sure_ that doing `parseInt` would throw but after re-testing it this doesn't seem to be the case. Will change. Thnx!